x-pack/test/api_integration/apis/transform/stop_transforms.ts

```diff
@@ -0,0 +1,182 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import expect from '@kbn/expect';
+
+import type { StopTransformsRequestSchema } from '../../../../plugins/transform/common/api_schemas/stop_transforms';
+import type { PutTransformsRequestSchema } from '../../../../plugins/transform/common/api_schemas/transforms';
+
+import { COMMON_REQUEST_HEADERS } from '../../../functional/services/ml/common_api';
+import { USER } from '../../../functional/services/transform/security_common';
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+import { asyncForEach, generateDestIndex, generateTransformConfig } from './common';
+
+export default ({ getService }: FtrProviderContext) => {
+  const esArchiver = getService('esArchiver');
+  const supertest = getService('supertestWithoutAuth');
+  const transform = getService('transform');
+
+  async function createAndRunTransform(transformId: string) {
+    // to be able to test stopping transforms,
+    // we create a slow continuous transform
+    // so it doesn't stop automatically.
+    const config: PutTransformsRequestSchema = {
+      ...generateTransformConfig(transformId),
+      settings: {
+        docs_per_second: 10,
+        max_page_search_size: 10,
+      },
+      sync: {
+        time: { field: '@timestamp' },
+      },
+    };
+
+    await transform.api.createAndRunTransform(transformId, config);
+  }
+
+  describe('/api/transform/stop_transforms', function () {
+    before(async () => {
+      await esArchiver.loadIfNeeded('ml/farequote');
+      await transform.testResources.setKibanaTimeZoneToUTC();
+    });
+
+    describe('single transform stop', function () {
+      const transformId = 'transform-test-stop';
+      const destinationIndex = generateDestIndex(transformId);
+
+      beforeEach(async () => {
+        await createAndRunTransform(transformId);
+      });
+
+      afterEach(async () => {
+        await transform.api.cleanTransformIndices();
+        await transform.api.deleteIndices(destinationIndex);
+      });
+
+      it('should stop the transform by transformId', async () => {
+        const reqBody: StopTransformsRequestSchema = [{ id: transformId, state: 'started' }];
+        const { body } = await supertest
+          .post(`/api/transform/stop_transforms`)
+          .auth(
+            USER.TRANSFORM_POWERUSER,
+            transform.securityCommon.getPasswordForUser(USER.TRANSFORM_POWERUSER)
+          )
+          .set(COMMON_REQUEST_HEADERS)
+          .send(reqBody)
+          .expect(200);
+
+        expect(body[transformId].success).to.eql(true);
+        expect(typeof body[transformId].error).to.eql('undefined');
+        await transform.api.waitForIndicesToExist(destinationIndex);
+      });
+
+      it('should return 200 with success:false for unauthorized user', async () => {
+        await transform.api.waitForTransformStateNotToBe(transformId, 'stopped');
```

Comment:
I think we don't need this here since it's already part of `createAndRunTransform` in the `before` code.

Reply:
Fixed in efcf9b3.